src/main/scala/karazin/scala/users/group/week5/homework/givens.scala

```diff
@@ -2,18 +2,22 @@ package karazin.scala.users.group.week5.homework
 
 object givens:
   
-  /* 
-    The trait is used for converting instances to a json string representation
-    Provide a type parameter(s) for the trait and the method 
-    and argument(s) and a result type
-  */ 
-  
-  trait DummyType
-  
-  trait JsonStringEncoder:
-    def encode: DummyType
+  trait JsonStringEncoder [T]:
+    def encode (toEncode: T): String
+
+  given IntToJson: JsonStringEncoder[Int] with
+    override def encode(toEncode: Int): String = toEncode.toString
+
+  given BoolToJson: JsonStringEncoder[Boolean] with
+    override def encode(toEncode: Boolean): String = toEncode.toString
+
+  given StringToJson: JsonStringEncoder[String] with
+    override def encode(toEncode: String): String = "\"" + toEncode + "\""
+
+  given ListEncoder[T] (using encoder: => JsonStringEncoder[T]): JsonStringEncoder[List[T]] with
```

Comment:
Can be simplified with `map` and https://www.scala-lang.org/api/2.13.3/scala/collection/immutable/List.html#mkString(start:String,sep:String,end:String):String